lib/tasks/disposable_domains.rake

```diff
@@ -0,0 +1,11 @@
+# rubocop:disable Rails/SkipsModelValidations
+require 'csv'
+namespace :disposable_domains do
+  task :load, %i[s3_url] => [:environment] do |_task, args|
+    file = Identity::Hostdata.secrets_s3.read_file(args[:s3_url])
+    csv_of_file = CSV.parse(file, headers: true)
```

Comment:
What's the process for creating this CSV file?

Alternatively, if [the original file is a `.txt`](https://github.com/fnando/email_data/blob/main/data/disposable_domains.txt) with one entry per line, could we read the file as-is with [`IO.readlines`](https://ruby-doc.org/core-2.1.0/IO.html#method-c-readlines) ?

I'm thinking it's a reasonable enough size (for a one-off script) to read the whole thing into memory, but alternatively we could batch this if necessary.

Reply:
the processing for creating this csv is just including the header and changing the extension to csv, so not overly complicated. and we could but we wouldn't be able to insert_all as easily into the database table.

Reply:
> the processing for creating this csv is just including the header and changing the extension to csv, so not overly complicated.

I'm primarily concerned with whether the process would be obvious to the next person who runs the script.

>and we could but we wouldn't be able to insert_all as easily into the database table.

It'd still be a pretty simple `map` though?

```rb
file = Identity::Hostdata.secrets_s3.read_file(args[:s3_url])
names = file.split("\n")
DisposableDomain.insert_all(names.map { |name| { name: } })
```